src/thunderbots/software/radio_communication/mrf/dongle.cpp

```diff
@@ -0,0 +1,610 @@
+#include "dongle.h"
+
+#include <sigc++/bind.h>
+#include <sigc++/functors/mem_fun.h>
+#include <sigc++/reference_wrapper.h>
+#include <unistd.h>
+
+#include <algorithm>
+#include <bitset>
+#include <cassert>
+#include <chrono>
+#include <cstdlib>
+#include <cstring>
+#include <exception>
+#include <iostream>
+#include <stdexcept>
+#include <string>
+#include <unordered_map>
+
+#include "constants.h"
+#include "radio_communication/visitor/mrf_primitive_visitor.h"
+#include "util/logger/init.h"
+
+namespace
+{
+    struct RadioConfig
+    {
+        uint8_t channel;
+        int symbol_rate;
+        uint16_t pan;
+    };
+
+    const RadioConfig DEFAULT_CONFIGS[4] = {
+        {25U, 250, 0x1846U},
+        {25U, 250, 0x1847U},
+        {25U, 250, 0x1848U},
+        {25U, 250, 0x1849U},
+    };
```

Comment:
What are these?

Reply:
In the old AI you can initialize the dongle with a different configuration (as cmdline args or through env variables), I don't specifically know what those do though other than setting a different channel to communicate on.

Reply:
Ah, could we just leave a quick comment to that effect?

Reply:
Yes these control the "channel" of the dongle. I think it technically just changes the PAN ID, not the channel bandwidth, but it's so you can run 2 donle's at once without conflicts (eg. someone mrf-testing and and someone controlling robots on the field).